api/db-cdr/generate-cdr/libproject/genomic_manifests.rb

```diff
@@ -582,17 +665,20 @@ def _maybe_update_preprod_access(project, all_tasks, add)
   preprod_deploy_account = must_get_env_value("all-of-us-rw-preprod", :publisher_account)
   preprod_ingest_bucket = must_get_env_value("all-of-us-rw-preprod", :accessTiers)["controlled"][:ingest_cdr_bucket]
 
-  if(project == "all-of-us-workbench-test")
+  if (project == "all-of-us-workbench-test")
     return
   end
 
   needs_cross_env_grant = false
+
   preprod_source_projects = (
     all_tasks
-      .map { |task| task["preprod_source_project"] }
+      .filter { |k, _v| k == "preprod_source_project" }
```

Comment:
is this a correct interpretation: works the same as before, but also filters to only the projects where `preprod_source_project` is defined?

That seems reasonable.  Does this PR require it, or it is cleanup?  (just curious, not blocking)

Reply:
the previous line didn't work because of nils as far as I remember when I made this change.